proposals/0371-isolated-synchronous-deinit.md

```diff
@@ -236,45 +310,94 @@ static void dealloc_impl_helper(void *ctx) {
 
 ### Exporting to Objective-C
 
-`deinit` isolation is relevant only when subclassing since Objective-C code cannot subclass Swift classes. The generated  `*-Swift.h` files contain no information about `deinit` isolation.
+`deinit` isolation is relevant only when subclassing. Since Objective-C code cannot subclass Swift classes, the generated  `*-Swift.h` files contain no addition information about new `deinit` features.
+
+### Interaction with ObjC runtime.
+
+All Objective-C-compatible Swift classes have `dealloc` method synthesized, which acts as a thunk to `__deallocating_deinit`. Normally when calling `super.deinit` from `__deallocating_deinit`, it is done by sending Objective-C `dealloc` message using `objc_msgSuper`.
+
+This ensures maximum possible compatibility with Objective-C, but comes with some runtime cost.
 
-### Isolated deinit of default actors
+For isolated synchronous `deinit`, this cost increases. For each class in the hierarchy `swift_task_deinitOnExecutor()` will be called, but slow path can be taken only for the most derived class. Other classes in the hierarchy would be doing redundant actor checks. To avoid this, isolated deinit bypasses Objective-C runtime when calling isolated `deinit` of Swift base classes, and calls `__isolated_deallocating_deinit` directly. Compatibility with Objective-C is preserved only on Swift<->Objective-C boundary in either direction.
```

Comment:
This is also a bit implementation-y, any way we could rephrase in terms of the actual language model?

Reply:
Reworded a bit, but it still contains mentions `__isolated_deallocating_deinit`.